include/swift/Basic/LangOptions.h

```diff
@@ -306,11 +306,14 @@ namespace swift {
       return CustomConditionalCompilationFlags;
     }
 
-    /// Whether our effective Swift version is in the Swift 3 family
-    bool isSwiftVersion3() const {
-      return EffectiveLanguageVersion.isVersion3();
+    /// Whether our effective Swift version is less than "X.Y"
+    bool isSwiftVersionLessThan(unsigned major, unsigned minor = 0) const {
+      return version::Version({major, minor}) >= EffectiveLanguageVersion;
     }
 
+    /// Whether our effective Swift version is in the Swift 3 family
+    bool isSwiftVersion3() const { return isSwiftVersionLessThan(4); }
```

Comment:
I would prefer to leave this exactly as is, unperturbed.  This means, revert isSwiftVersion3 to what it was and only add new code for the isSwiftversionLessThen function.